WordPress/Classes/Services/RawBlockEditorSettingsService.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import WordPressKit
+import WordPressShared
+
+class RawBlockEditorSettingsService {
+    private let blog: Blog
+    private let remoteAPI: WordPressOrgRestApi
+    private var isRefreshing: Bool = false
+
+    init?(blog: Blog) {
+        guard let remoteAPI = WordPressOrgRestApi(blog: blog) else {
+            return nil
+        }
+
+        self.blog = blog
+        self.remoteAPI = remoteAPI
+    }
+
+    @MainActor
+    private func fetchSettingsFromAPI() async throws -> [String: Any] {
+        let result = await self.remoteAPI.get(path: "/wp-block-editor/v1/settings")
+        switch result {
+        case .success(let response):
+            guard let dictionary = response as? [String: Any] else {
+                throw NSError(domain: "RawBlockEditorSettingsService", code: 1, userInfo: [NSLocalizedDescriptionKey: "Invalid response format"])
+            }
+            blog.rawBlockEditorSettings = dictionary
+            return dictionary
+        case .failure(let error):
+            throw error
+        }
+    }
+
+    @MainActor
+    func fetchSettings() async throws -> [String: Any] {
```

Comment:
What happens if we've never fetched the settings and the network connection fails?

In my testing it _seems_ like it'll just show a spinner forever?

If we don't have locally cached settings and the network request fails we should probably:

1. Try again some number of times (maybe 2?)
2. Show an error message 

I could see an argument for showing _some_ kind of editor with no remote settings, but I don't know how feasible that is?

Reply:
Your experience is surprising. These changes were built to accomplish what you describe. When I block the network request (via Charles Proxy) for the editor settings, the editor continues to load with default editor settings after a moment. 

How are you causing the network connection failure in your testing? 

The screen recording below showcases both the default settings and site-specific settings. 

https://github.com/user-attachments/assets/489279e2-3213-4468-8ad2-7df8ae858c1b

Reply:
In fact, the situation you describe is a part of the testing instructions in the PR description. 😄 I'm perplexed as to why you may be experiencing a different outcome.